concordium-contracts-common-derive/src/derive.rs

```diff
@@ -6,14 +6,42 @@ use quote::ToTokens;
 use std::{collections::HashMap, convert::TryFrom, ops::Neg};
 use syn::{spanned::Spanned, DataEnum, Ident, Meta};
 
+macro_rules! ensure {
+    ($condition: expr, $span: expr, $message: expr) => {
+        if !$condition {
+            bail!($span, $message);
+        }
+    };
+    ($condition: expr, $span: expr, $message: expr, $($arg:tt)*) => {
+        if !$condition {
+            bail!($span, $message, $($arg)*);
+        }
+    };
+}
+
+macro_rules! bail {
+    ($span: expr, $message: expr) => {
+        return Err(syn::Error::new(
+            $span,
+            format!($message),
```

Comment:
I did not consider the case of something other than string literals, but I think that is fine since this is just used internally.
The reason for using format here is that you can inline the variables `format!("{x}")` as opposed to `format!("{}",x)`.